zulipterminal/model.py

```diff
@@ -327,6 +329,29 @@ def exception_safe_result(self, future: 'Future[bool]') -> bool:
         except zulip.ZulipError:
             return False
 
+    def fetch_all_topics(self, workers: int) -> None:
+        """
+        Distribute stream ids across threads in order to fetch
+        topics concurrently.
+        """
+        with ThreadPoolExecutor(max_workers=workers) as executor:
+            list_of_streams = list(self.stream_dict.keys())
+            thread_objects = {}  # type: Dict[int, Future[bool]]
+            thread_objects = {i: executor.submit(self.get_topics_in_stream,
+                                                 list_of_streams[i::workers])
+                              for i in range(workers)}
+        # FIXME: This handles thread safety. But causes an extra overhead of
+        # ~ 4 seconds as has to wait for threads to complete.
+            # wait(thread_objects.values())
+        results = {
+            str(name): self.exception_safe_result(thread_object)
+            for name, thread_object in thread_objects.items()
+        }  # type: Dict[str, bool]
+        if not all(results.values()):
+            failures = ['Thread'+str(name) for name,
+                        result in results.items() if not result]
+            raise ServerConnectionFailure(", ".join(failures))
```

Comment:
I'm not sure we really mind which thread failed here?
(adding this as a single comment, since for some reason GH generated multiple reviews already)

Reply:
All right. Wanted to have a bit more discussion regarding this.

Reply:
Updated this based on the discussion. :+1: